src/GraphQL/GraphQLExtensions.cs

```diff
@@ -16,9 +16,30 @@ public static class GraphQLExtensions
     {
         private static readonly Regex _trimPattern = new Regex("[\\[!\\]]", RegexOptions.Compiled);
 
+        private static readonly char[] _bangs = new char[] { '!', '[', ']' };
+
         public static string TrimGraphQLTypes(this string name)
         {
-            return _trimPattern.Replace(name, string.Empty).Trim();
+            name = name.Trim();
+
+            if (name.Length <= 256)
+            {
+                Span<char> buffer = stackalloc char[name.Length];
+
+                int index = 0;
+                for (int i=0; i<name.Length; ++i)
+                {
+                    char c = name[i];
+                    if (!_bangs.Contains(c))
+                        buffer[index++] = c;
+                }
+
+                return index == name.Length ? name : buffer.Slice(0, index).ToString();
+            }
+            else
+            {
+                return _trimPattern.Replace(name, string.Empty);
```

Comment:
I decided not to change this code branch at all. Hardly anyone has such long names.